action.yml

```diff
@@ -37,7 +37,7 @@ runs:
   steps:
     - name: Commit Range
       id: commit-range
-      uses: ./.github/actions/commit-range
+      uses: akaihola/darker/.github/actions/commit-range@1.7.1
```

Comment:
[Question] Is `@1.7.1` here necessary?

Reply:
Good point. Would omitting it use the action from `@main`? I'll try to find out.

Reply:
From [`jobs.<job_id>.uses` documentation]](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iduses):

>- `{owner}/{repo}/.github/workflows/{filename}@{ref}` for reusable workflows in public and private repositories.
>
>[...]
>`{ref}` can be a SHA, a release tag, or a branch name. Using the commit SHA is the safest for stability and security. For more information, see "[Security hardening for GitHub Actions](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#reusing-third-party-workflows)."

All examples on that documentation page do contain a `@{ref}`.

Using `@main` would cause old versions of the action to call `commit-range` from current `@main`, so effectively the action version specifier wouldn't guarantee that all the code which is run will always stay the same.

Thus, I think using a version number here is the proper thing to do. Or might I be missing something?

Reply:
@akaihola `1.7.1` was the version that has the issue if I am not mistaken. So it might be better if we use the proper version after the release.

Reply:
@roniemartinez actually the `commit-range` action wasn't modified in Darker 1.7.1. The only problem is pointing to the action using a same-repository path instead of a full path. This works only in Darker's own workflows, but not when using the Darker action in other repositories.

Fixing this in the `master` branch after each release doesn't feel right.

We could of course keep pointing to a past version (`@1.6.0`, `@1.7.0` and `@1.7.1` are identical) and deal with this dilemma only when there's a need to actually modify `.github/actions/commit-range/`.